### PR TITLE
post about map puts shorthand in elixir

### PR DESCRIPTION
**Pull request by outragedpinkracoon**

### CODE REVIEW

**Review by picandocodigo — APPROVED** [review]

File: _posts/2018-12-19-map-puts-shorthand.markdown [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+layout: post
+title:  "Elixir: Map Puts Shortand"
+date:   2018-12-17 07:37:29 +0000
+categories: elixir
+author: Valerie Dryden
+---
+
+In Elixir, we can use the `Map.puts` function to take an existing map and return a new map with either an updated value for a key, or to add a new key / value pair.
+
+cats = %{barnaby: 'red', boba: 'white', max: 'black'}
```
Comment (munckymagik) [inline _posts/2018-12-19-map-puts-shorthand.markdown:12]: Question: do we have code blocks enabled in this blog yet? If so tick, tick, tick!

Comment (outragedpinkracoon) [reply _posts/2018-12-19-map-puts-shorthand.markdown:12]: Oops! Good catch :) Have updated.

File: _posts/2018-12-19-map-puts-shorthand.markdown [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+```elixir
+cats = Map.put(cats, :max, 'grey')
+```
+
+Or we can use the nice shorthand syntax to do the same thing:
+```elixir
+cats = %{ cats | max: 'grey' }
+```
+
+Note this only works when updating an existing value in a map, not to insert a new key / value. To insert a new key / value we still need to use Map.put.
```
Comment (paulanthonywilson) [inline _posts/2018-12-19-map-puts-shorthand.markdown]: Suggestion: remove some ambiguity with `s/this/the shorthand syntax/`

**Review by paulanthonywilson — APPROVED** [review]

File: _posts/2018-12-19-map-puts-shorthand.markdown (hunk 2 above)
Comment (paulanthonywilson) [reply _posts/2018-12-19-map-puts-shorthand.markdown]: What i mean is say "Note that the shorthand syntax ..."

Comment (outragedpinkracoon) [reply _posts/2018-12-19-map-puts-shorthand.markdown]: Nice, have updated.

**Pull request merged**